lib/idx/remediate.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable max-statements */
+/* eslint-disable complexity */
+import { IdxResponse, isRawIdxResponse, RemediationValues, Remediator, IdxRemediation } from '../types';
+import { createApiError, isErrorResponse } from './util';
+import Identify from './remediatiors/Identify';
+import ChallengeAuthenticator from './remediatiors/ChallengeAuthenticator';
+
+const REMEDIATORS = {
```

Comment:
I feel `REMEDIATORS` should be defined in a specific method, like `authenticate` as it provides action paths for a certain flow to try.

Reply:
It's a cheap version of a "view factory". Just a simple map acting as a registry. It could go in a separate file but currently only being used here. As currently implemented, it is a constant, providing a lookup from `name => type`. If we need some dynamic logic then it would need to evolve, probably to a factory pattern.

Reply:
With the current assumption (policy has been pre-defined), and each flow (method) is bind to some specific `REMEDIATORS`, I think it would make sense to couple method and remediators together so method do not run into unsupported `REMEDIATORS` due to things like policy change.